projects/manager/src/app/session/session.service.ts

```diff
@@ -0,0 +1,119 @@
+import {Injectable} from '@angular/core';
+import {Configuration, DefaultConfig, SessionApi, UserSession} from '@geoengine/openapi-client';
+import {Observable, ReplaySubject, filter, first, from, map, tap} from 'rxjs';
+
+const PATH_PREFIX = window.location.pathname.replace(/\//g, '_').replace(/-/g, '_');
+
+@Injectable({
+    providedIn: 'root',
+})
+export class SessionService {
+    protected readonly session$ = new ReplaySubject<UserSession | undefined>(1);
+
+    protected sessionInitialized = false;
+
+    constructor() {
+        this.session$.subscribe((session) => {
+            // storage of the session
+            this.saveSessionInBrowser(session);
+        });
+
+        this.restoreSessionFromBrowser().subscribe({
+            next: (session) => {
+                this.session$.next(session);
+            },
+            error: () => {
+                this.session$.next(undefined);
+            },
+        });
```

Comment:
What would be the race condition? saving and loading should not interfere here I think.

Reply:
how could that happen. The first subscription saves whatever session gets `next`ed in the browser. And the second subscription `next`s whatever is stored in the browser. So it will just read from the browser and store it back again. We could skip(1) but does it really matter?